### PR TITLE
code-health: Migrating hardcoded-strings

### LINKED ISSUE #379: Migrate hardcoded strings to strings.xml value file
**Is your feature request related to a problem? Please describe.**
Strings in value file, make for easy changes in UI, reusability etc.

**Describe the solution you'd like**
Migrate the hardcoded strings to strings.xml file.

**Describe alternatives you've considered**
N/A

**Additional context**
One example file includes:https://github.com/google/android-fhir/blob/master/core/src/main/java/com/google/android/fhir/resource/Resources.kt
---> Line 30-37

**Would you like to work on the issue?**
Yes.

### PR DESCRIPTION
**Pull request by Ana2k**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straighforward changes).**

Fixes #379

**Description**
Migrate user-visible hard-coded string files

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: |Code-Health| 

**Screenshots (if applicable)**
N/A

**Checklist**
- [x] I have read and acknowledge the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read [How to Contribute](https://github.com/google/android-fhir/blob/master/docs/contributing.md)
- [x] I have read the [Developer's guide](https://github.com/google/android-fhir/wiki/Developer's-Guide)
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

**Comment by Manikant25** [comment]
@Ana2k  Try to run  `./gradlew spotlessApply` and  `./gradlew spotlessCheck`. I think there is a problem in the coding style.

**Comment by Ana2k** [comment]
> @Ana2k Try to run `./gradlew spotlessApply` and `./gradlew spotlessCheck`. I think there is a problem in the coding style.

@Manikant25 thanks for the advice, its currently WIP(work-in-progress). After I complete making all the changes I will surely. Thanks :)

File: reference/src/main/java/com/google/android/fhir/reference/data/SamplePatients.kt [hunk 1]
```diff
@@ -111,10 +113,10 @@ class SamplePatients {
     // Show nothing if no values available for datetime and value quantity.
     val dateTimeStr =
       if (observation.hasEffectiveDateTimeType()) observation.effectiveDateTimeType.asStringValue()
-      else "No effective DateTime"
+      else Resources.getSystem().getString(R.string.no_effective_date_time)
     val value =
       if (observation.hasValueQuantity()) observation.valueQuantity.value.toString()
-      else "No ValueQuantity"
+      else Resources.getSystem().getString(R.string.no_value_quantity)
```
Comment (epicadk) [inline reference/src/main/java/com/google/android/fhir/reference/data/SamplePatients.kt:119]: You can only access system resources like this not application resources.

**Comment by epicadk** [comment]
> > @Ana2k Try to run `./gradlew spotlessApply` and `./gradlew spotlessCheck`. I think there is a problem in the coding style.
> 
> @Manikant25 thanks for the advice, its currently WIP(work-in-progress). After I complete making all the changes I will surely. Thanks :)

You can convert it to draft so people will know you are still working on it. No need to add WIP in the title.

File: cqlreference/src/main/java/com/google/android/fhir/cqlreference/CqlLoadActivity.kt [hunk 2]
```diff
@@ -96,13 +96,13 @@ class CqlLoadActivity : AppCompatActivity() {
         fhirEngine.save(resource)
         Snackbar.make(
             cqlLibraryUrlInput,
-            "Loaded " + resource.resourceType.name + " with ID " + resource.id,
+            getString(R.string.snackbar_loaded) + resource.resourceType.name + getString(R.string.snackbar_with_Id) + resource.id,
```
[the remark this answers is not in the document]
Comment (Ana2k) [reply cqlreference/src/main/java/com/google/android/fhir/cqlreference/CqlLoadActivity.kt:99]: - 

- Please have a look its

> > > @Ana2k Try to run `./gradlew spotlessApply` and `./gradlew spotlessCheck`. I think there is a problem in the coding style.
> > 
> > 
> > @Manikant25 thanks for the advice, its currently WIP(work-in-progress). After I complete making all the changes I will surely. Thanks :)
> 
> You can convert it to draft so people will know you are still working on it. No need to add WIP in the title.

Afaik WIP means Work In Progress, and I feel more comfortable doing this as seen in other open source orgs. Thanks for the suggestion!

**Comment by Ana2k** [comment]
> > > @Ana2k Try to run `./gradlew spotlessApply` and `./gradlew spotlessCheck`. I think there is a problem in the coding style.
> > 
> > 
> > @Manikant25 thanks for the advice, its currently WIP(work-in-progress). After I complete making all the changes I will surely. Thanks :)
> 
> You can convert it to draft so people will know you are still working on it. No need to add WIP in the title.

Afaik WIP means Work In Progress, and I feel more comfortable doing this as seen in other open source orgs. Thanks for the suggestion!

**Comment by Ana2k** [comment]
@jingtang10 I have finalised this pr. Please review it.

File: cqlreference/src/main/java/com/google/android/fhir/cqlreference/CqlLoadActivity.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ana2k) [reply cqlreference/src/main/java/com/google/android/fhir/cqlreference/CqlLoadActivity.kt]: Please check now, the pr has been finalised.

**Comment by epicadk** [comment]
> Afaik WIP means Work In Progress, and I feel more comfortable doing this as seen in other open source orgs. Thanks for the suggestion!

It was suggested by one of the maintainers here. : /  
https://github.com/google/android-fhir/pull/386#issuecomment-805576021

**Comment by Ana2k** [comment]
> > Afaik WIP means Work In Progress, and I feel more comfortable doing this as seen in other open source orgs. Thanks for the suggestion!
> 
> It was suggested by one of the maintainers here. : /
> [#386 (comment)](https://github.com/google/android-fhir/pull/386#issuecomment-805576021)

Yupp saw it. Thanks! :)

**Review by jingtang10 — CHANGES_REQUESTED** [review]
1. please don't put exception strings in resource files.
2. please revert all changes in `cqlreference` module since the ui will change.

**Comment by jingtang10** [comment]
this is not a feature. this is code health. please update PR type and title.

**Comment by Ana2k** [comment]
> 2\. all changes in `cqlreference` module since the ui will change.

Snackbar changes?(they are in cqlreference)

**Comment by Ana2k** [comment]
> 1. please don't put exception strings in resource files.
> 2. please revert all changes in `cqlreference` module since the ui will change.

1. I can turn translatable as false for the exception message.
Then I hope no issue with i10 or i18 translation will be there?
2. Should I revert the Snackbars too?(they are in cqlreference non UI part)

**Comment by jingtang10** [comment]
@Ana2k apologies but I'm going to close this because we're going to rewrite the UIs. Thanks again for your contribution!

**Pull request closed without merging**